Sources/StreamChat/Config/BaseURL.swift

```diff
@@ -46,6 +46,23 @@ public struct BaseURL: CustomStringConvertible {
         }
         
         urlString = urlString.trimmingCharacters(in: CharacterSet(charactersIn: "/"))
+
+        // Leverages mocked web socket server when running the UI tests.
+        // Two conditions need to be met in order to leverage the web socket server.
+        //   1. App runs in Debug build configuration
+        //   2. `USE_MOCK_SERVER` is set as launch argument
+        #if DEBUG
+        if ProcessInfo.processInfo.arguments.contains("USE_MOCK_SERVER") {
+            let env = ProcessInfo.processInfo.environment
```

Comment:
Maybe move this code in a separate private method?

Reply:
@martinmitrevski this logic happens within constructor.  Both `restAPIBaseURL` and `webSocketBaseURL` needs to be set before the `BaseURL` is fully initialized.

One way to leverage `private func` for this logic would be by turning the `restAPIBaseURL` and `webSocketBaseURL` into readonly force unwrapped vars. I would like to avoid that :) 

Let me know your thoughts ;)

Reply:
also not a fan of force unwrapping them. :) Well, in that case, either make the private func static or leave everything as it is. In any case, not a major thing to spend time on.

Reply:
Thanks. Ill keep this discussion open, it's a good comment - I also spent some time playing around alternatives prior opening this PR == Im open for suggestions :)

Reply:
Maybe sth like this, but not sure if it brings any additional value: 
```swift
        #if DEBUG
        if ProcessInfo.processInfo.arguments.contains("USE_MOCK_SERVER") {
            let mockServerUrls = Self.mockServerUrls(with: urlString)
            restAPIBaseURL = mockServerUrls.restAPIBaseURL
            webSocketBaseURL = mockServerUrls.webSocketBaseURL
            return
        }
        #endif
```

```swift
private static func mockServerUrls(with urlString: String) -> (restAPIBaseURL: URL, webSocketBaseURL: URL) {
        let env = ProcessInfo.processInfo.environment
        let httpHost = env["MOCK_SERVER_HTTP_HOST"] ?? "https://\(urlString)/"
        let websocketHost = env["MOCK_SERVER_WEBSOCKET_HOST"] ?? "wss://\(urlString)/"
        let port = env["MOCK_SERVER_PORT"] ?? "443"
        let restAPIBaseURL = URL(string: "\(httpHost):\(port)/")!
        let webSocketBaseURL = URL(string: "\(websocketHost):\(port)/")!
        return (restAPIBaseURL: restAPIBaseURL, webSocketBaseURL: webSocketBaseURL)
    }
```